src/Controller/EnMarche/AdherentMessage/CommitteeMessageController.php

```diff
@@ -2,17 +2,263 @@
+            ]);
+        }
+
+        return $this->renderTemplate('message/update.html.twig', [
+            'form' => $form->createView(),
+            'committee' => $committee,
+            'route_params' => ['committee_slug' => $committee->getSlug()],
+        ]);
+    }
+
+    /**
+     * @Route("/{uuid}/filtrer", name="filter", methods={"GET"})
+     *
+     * @Security("is_granted('IS_AUTHOR_OF', message)")
+     */
+    public function filterMessageAction(
+        Request $request,
+        AbstractAdherentMessage $message,
+        FilterFormFactory $formFactory,
+        ObjectManager $manager,
+        Committee $committee = null
+    ): Response {
+        $this->disableInProduction();
+
+        if ($message->isSent()) {
+            throw new BadRequestHttpException('This message has been already sent.');
+        }
+
+        return $this->renderTemplate('message/filter/committee.html.twig', [
+            'message' => $message,
+            'committee' => $committee,
+            'route_params' => ['committee_slug' => $committee->getSlug()],
+        ]);
+    }
+
+    /**
+     * @Route("/{uuid}/visualiser", name="preview", methods={"GET"})
+     *
+     * @Security("is_granted('IS_AUTHOR_OF', message)")
+     */
+    public function previewMessageAction(AbstractAdherentMessage $message, Committee $committee = null): Response
+    {
+        $this->disableInProduction();
+
+        if (!$message->isSynchronized()) {
+            throw new BadRequestHttpException('Message preview is not ready yet.');
+        }
+
+        return $this->renderTemplate('message/preview.html.twig', [
+            'message' => $message,
+            'committee' => $committee,
+            'route_params' => ['committee_slug' => $committee->getSlug()],
+        ]);
+    }
+
+    /**
+     * @Route("/{uuid}/supprimer", name="delete", methods={"GET"})
+     *
+     * @Security("is_granted('IS_AUTHOR_OF', message)")
+     */
+    public function deleteMessageAction(
+        AbstractAdherentMessage $message,
+        ObjectManager $manager,
+        Committee $committee = null
+    ): Response {
+        $this->disableInProduction();
+
+        $manager->remove($message);
+        $manager->flush();
+
+        $this->addFlash('info', 'adherent_message.deleted_successfully');
+
+        return $this->redirectToMessageRoute('list', ['committee_slug' => $committee->getSlug()]);
+    }
+
+    /**
+     * @Route("/{uuid}/send", name="send", methods={"GET"})
+     *
+     * @Security("is_granted('IS_AUTHOR_OF', message)")
+     */
+    public function sendMessageAction(
+        AbstractAdherentMessage $message,
+        Manager $manager,
+        ObjectManager $entityManager,
+        Committee $committee = null
+    ): Response {
+        $this->disableInProduction();
+
+        if (!$message->isSynchronized()) {
+            throw new BadRequestHttpException('The message is not yet ready to send.');
```

Comment:
`'The message is not ready to send yet.` ?

Reply:
changed